File: TestCase/TestCase/Models/Transaction/TransactionModel.cs
using TestCase.Entities;
using static TestCase.Entities.Enums;

namespace TestCase.Models.Transaction
{
    public class TransactionModel
    {
        #region Properies

        public int Id { get; set; }
        public string Status { get; set; }
        public string Type { get; set; }
        public string ClientName { get; set; }
        public decimal Amount { get; set; }

        #endregion

        #region Constructors

        public TransactionModel()
        { 
        
        }

        public TransactionModel(TransactionEntity entity)
        {
            Id = entity.Id;
            Status = entity.Status.ToString();
            Type = entity.Type.ToString();
            ClientName = entity.ClientName;
            Amount = entity.Amount;
        }

        #endregion
    }
}

File: TestCase/TestCase/ClientApp/src/app/models/transaction/transaction.model.ts
export class TransactionModel {
  public id: number;
  public status: string;
  public type: string;
  public clientName: string;
  public amount: number;

  constructor() {
    this.id = null;
    this.status = "";
    this.type = "";
    this.clientName = "";
    this.amount = null;
  }
}
